pkg/kubelet/pleg/generic.go

```diff
@@ -122,34 +165,66 @@ func (g *GenericPLEG) relist() {
 		return
 	}
 
+	// Build a new map.
+	podMap := buildPodMap(pods)
```

Comment:
I think we can avoid creating a new map for each relisting here by maintaining just two maps, and reuse one map in each iteration.